pkg/cloudops/aws/aws.go

```diff
@@ -0,0 +1,257 @@
+package aws
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/libopenstorage/openstorage/pkg/cloudops"
+	"github.com/libopenstorage/openstorage/pkg/util"
+	"github.com/sirupsen/logrus"
+)
+
+// ErrAWSEnvNotAvailable is the error type when aws credentials are not set
+var ErrAWSEnvNotAvailable = fmt.Errorf("AWS credentials are not set in environment")
+
+type awsOps struct {
+	instInfo    *cloudops.InstanceInfo
+	ec2         *ec2.EC2
+	autoscaling *autoscaling.AutoScaling
+}
+
+// NewAWSps returns an instance of the AWS cloud ops implementation
+func NewAWSOps() (cloudops.Ops, error) {
+	ops := &awsOps{
+		instInfo: &cloudops.InstanceInfo{},
+	}
+	err := ops.populateBasicInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	ops.instInfo.Region = ops.instInfo.Zone[:len(ops.instInfo.Zone)-1]
+
+	ops.ec2 = ec2.New(
```

Comment:
We need to have a New(), that takes in a Credentials object, so that we support instance privileges as well and not just Env variables. We do something similar in storageops.

Reply:
in clouds, I'm updating NewAWSOps to be consistent with how GCE works. This will use the credentials package from libopenstorage/secrets.